Content.Server/Forensics/Systems/ForensicsSystem.cs

```diff
@@ -70,28 +70,29 @@ private void OnMeleeHit(EntityUid uid, ForensicsComponent component, MeleeHitEve
             }
         }
 
-        private void OnAfterInteract(EntityUid uid, ForensicsComponent component, AfterInteractEvent args)
+        private void OnAfterInteract(EntityUid uid, CleansForensicsComponent component, AfterInteractEvent args)
         {
             if (args.Handled)
                 return;
 
-            if (!_tagSystem.HasTag(args.Used, "CleansForensics"))
+            if (!TryComp<ForensicsComponent>(args.Target, out var forensicsComp))
                 return;
 
-            if((component.DNAs.Count > 0 && component.CanDnaBeCleaned) || (component.Fingerprints.Count + component.Fibers.Count > 0))
+            if((forensicsComp.DNAs.Count > 0 && forensicsComp.CanDnaBeCleaned) || (forensicsComp.Fingerprints.Count + forensicsComp.Fibers.Count > 0))
             {
-                var doAfterArgs = new DoAfterArgs(EntityManager, args.User, component.CleanDelay, new CleanForensicsDoAfterEvent(), uid, target: args.Target, used: args.Used)
+                var doAfterArgs = new DoAfterArgs(EntityManager, args.User, forensicsComp.CleanDelay, new CleanForensicsDoAfterEvent(), uid, target: args.Target, used: args.Used)
```

Comment:
might as well put the delay on CleansForensicsComponent, then syndie soap can trivially be buffed

Reply:
For sure. I thought about doing that, but couldn't think of a case where one would have a different time. Syndie soap would be faster, good idea.